vmdb/app/controllers/application_controller/explorer.rb

```diff
@@ -927,8 +927,7 @@ def x_get_tree_custom_kids(object, options)
       return count_only ? objects.length : objects
     when :alert_profile
       # Add all alert profiles so links back from Alerts etc will work - TODO: figure out how to load on demand
-      objects = MiqAlertSet.all(:conditions=> ["(mode = ?)", object[:id].split('-')],
-                          :order => "lower(description) ASC")
+      objects = MiqAlertSet.where(:mode => object[:id].split('-')).order("lower(description) ASC")
```

Comment:
My other argument is simpler: the differences between array and relation are in the mutation methods.

And any far-away code that's directly mutating an "array" (or any other object) it got handed, without making its own copy -- and without said mutation being the sum total of its API contract -- is Doing The Wrong Thing.